cpp/arcticdb/storage/storage.hpp

```diff
@@ -234,9 +251,10 @@ class Storage {
     // the predicate.
     virtual bool do_iterate_type_until_match(KeyType key_type, const IterateTypePredicate& visitor, const std::string & prefix) = 0;
 
-    [[nodiscard]] virtual ObjectSizes do_get_object_sizes([[maybe_unused]] KeyType key_type, [[maybe_unused]] const std::string& prefix) {
+    virtual void do_visit_object_sizes([[maybe_unused]] KeyType key_type, [[maybe_unused]] const std::string& prefix,
```

Comment:
nit: if we remove the parameter names we can also omit `[[maybe_unused]]` which makes it a bit shorter.